### PR TITLE
Add GitHub deployments for pr previews

### LINKED ISSUE #18769: Leverage the GitHub deployments for PR mirroring
@jugglinmike has been implementing a PR-mirroring behaviour for web-platform-tests.live.

This technique involves a special label, `pull-request-has-preview`, but there's no link provided to the user on the PR itself. It should be a small leap to get the GitHub action to [create a deployment via the GitHub API](https://developer.github.com/v3/repos/deployments/#create-a-deployment) and then set the status when the action completes.

### PR DESCRIPTION
**Pull request by lukebjerring**
Fixes #18769

### CODE REVIEW

**Comment by lukebjerring** [comment]
@jugglinmike - I've put `wpt.reviews` as a placeholder here; where are mirrors currently being put?

Also, how do I test this code..?

File: tools/ci/update_pr_preview.py [hunk 1]
```diff
@@ -177,6 +178,69 @@ def remove_label(self, pr_number, label_name):
...
+            logger.error('Failed to created deployment')
+            return
+
+        logger.info('Creating status for deployment {}'.format(
+            deployment.id
+        ))
+        url = url + '/{}/statuses'
+        data = {
+            'state': 'success',
+            'environment_url': 'https://wpt.reviews/{}/'.format(
```
Comment (lukebjerring) [inline tools/ci/update_pr_preview.py]:
```suggestion
            'environment_url': 'http://w3c-test.org/submissions/{}'.format(
```

As per https://github.com/web-platform-tests/wpt/pull/18772/files#diff-fad1e181b4cb0da669a8e5a28002069fR49

**Comment by jugglinmike** [comment]
[Tests for the script are defined here.](https://github.com/web-platform-tests/wpt/blob/fb9b5a1d816ab6e90f20534d6f3282f82075e460/tools/ci/tests/test_update_pr_preview.py) The script accepts the domain as input (since one might ostensiby want to run it against a GitHub Enterprise instance), and the tests use this to redirect requests to the loopback interface. In other words, you can script the client/server protocol.

**Comment by lukebjerring** [comment]
Closing this stale PR.

**Pull request closed without merging**